### DISCUSSION TITLE
Widevine support

### DISCUSSION THREAD

**Discussion by rahul-kurup** [Joshua-Riek/ubuntu-rockchip#44] [Ideas]
Hey do you have future plans to implement widevine support (or a workaround of sorts) so that we can play Netflix/Prime Video directly in Chromium.

**Comment by Joshua-Riek** [comment]
I have not heard of widevine before, but I will look at it. Being able to stream videos on a platform like Netflix would be awesome.

**Reply by rahul-kurup** [reply]
The thing is widevine comes with the native bundle of Google Chrome (stable). But as Chrome is not available for arm64, we have to use Chromium which does not have widevine. You may search for "libwidevinecdm0" for more info. I tried importing the said lib on armbian jammy (op5) using my limited knowledge of linux; but it failed. 
This library (widevine) was added to raspberry os, a year ago

**Reply by ferrellsl** [reply]
If you want/need Widevine support on your Orange Pi 5, just install the Brave browser.  It support Widevine and is based on Chrome so it supports all the usual Chrome plug-ins and extensions as well.  To install it, go here:  https://brave.com/linux/

To test that Widevine is working properly, go here and play these videos to confirm it's working properly:  https://shaka-player-demo.appspot.com/

**Reply by Joshua-Riek** [reply]
That might work, but it won't have proper hardware acceleration for smooth video playback.

**Reply by rahul-kurup** [reply]
@ferrellsl Hey I have tried Brave and Vivaldi before. They simply won't run like that. For instance, when you play something on:

1. Netflix -  it throws the error "Pardon the interruption....."
2. Prime Video - it throws the error "Digital Rights Error. Your web browser is missing widevinecdm"

This is because none of the browsers comes with widevinecdm by default for arm architecture.

**Reply by ferrellsl** [reply]
It seems to be hit and miss.  I can play Widevine videos on some sites and not on others.  For example, this site works fine: https://www.theoplayer.com/theoplayer-drm-aes-128-encryption

And some sites play DRM protected videos but they have artifacts and others say that I don't have a supported key system when I try to play them.

I don't have NetFlix but let me see what kind of errors get thrown when I try to play something from Amazon Prime and get back to you.

**Reply by ferrellsl** [reply]
Yep, I get an error about WidevineCDM missing.

**Reply by rahul-kurup** [reply]
@Joshua-Riek Did u get a chance to check this out?

**Reply by Joshua-Riek** [reply]
I have not yet, I have been busy with work and testing out some new GPU drivers and a more recent chromium version on the opi5. Ill be able to take a closer look this Friday.

**Reply by rahul-kurup** [reply]
Oh great. I'm trying some workarounds for this. If it works I'll post an update here.

**Comment by rahul-kurup** [comment]
Spotify (open.spotify.com) also doesn't load cuz of DRM issue :(

**Reply by Joshua-Riek** [reply]
I have been looking into this and have had no luck so far.

**Reply by rahul-kurup** [reply]
https://archive.raspberrypi.org/debian/pool/main/w/widevine/

These are the widevine packages for raspberry os. Not sure whether they can be reverse engineered for OrangePi

**Comment by Joshua-Riek** [comment]
Widevine has arrived on the Rock 5B and should also be able to work on the Orange Pi 5. I will test this out when I have time. https://youtu.be/QwyRgisFe5Q

**Reply by Joshua-Riek** [reply]
Spotify and Amazon Prime video working!
![Screenshot from 2023-03-22 18-13-20](https://user-images.githubusercontent.com/10427125/227053019-7b200116-41a8-4e23-9226-9c640f7cb0c4.png)
![Screenshot from 2023-03-22 18-13-57](https://user-images.githubusercontent.com/10427125/227053021-f1a32cfa-fdcb-4f09-aeaf-82b20f50c7aa.png)
![Screenshot from 2023-03-22 18-25-52](https://user-images.githubusercontent.com/10427125/227053024-3564cc6f-c5f6-41f4-8a4f-06f3ea3632e3.png)

**Reply by ironiris** [reply]
WOW

**Reply by Joshua-Riek** [reply]
Netflix does not work, but it's incredible to see progress!

**Reply by rahul-kurup** [reply]
> Netflix does not work, but it's incredible to see progress!

For Netflix, this can be done:

<img width="657" alt="image" src="https://user-images.githubusercontent.com/29568606/227257915-0e821f51-1821-4a9d-a804-3225eece4c96.png">

Source: https://forum.radxa.com/t/lets-get-widevine-working/15391/26

**Reply by rahul-kurup** [reply]
Can you share the commands you executed to enable this on your system, I'd like to try it; before the next release

**Comment by rahul-kurup** [comment]
@Joshua-Riek Can you add this in the next release, if possible?

**Reply by Joshua-Riek** [reply]
Yes, I will be adding it in the next release!

**Comment by nrpetonr** [comment]
use this [script ](https://github.com/AsahiLinux/widevine-installer)followed by [browser spoofing,](https://chromewebstore.google.com/detail/user-agent-switcher-and-m/bhchdcejhohfmigjafbampogmaanbfkg?hl=en) to fix the problem with drm sources, even on 24.04. enjoy...

**Reply by slashxzero** [reply]
But which version should I use?
thank you.

**Reply by nrpetonr** [reply]
not sure what u mean by your question
